src/main/java/com/ays/assignment/repository/AssignmentRepository.java

```diff
@@ -33,6 +34,16 @@ public interface AssignmentRepository extends JpaRepository<AssignmentEntity, St
      */
     Optional<AssignmentEntity> findByUserIdAndStatusNot(String userId, AssignmentStatus status);
 
+    /**
+     * Retrieves an optional AssignmentEntity based on the provided USER ID and statuses
+     *
+     * @param userId   The unique identifier of the user for whom the assignment is being checked.
+     * @param statuses A collection of Assignment Statuses
+     * @return An optional AssignmentEntity that matches with the specified user ID and
+     * includes at least one of the specified Assignment statuses
+     */
+    Optional<AssignmentEntity> findByUserIdAndStatusIn(String userId, Collection<AssignmentStatus> statuses);
```

Comment:
`EnumSet` gönderdiğimiz için burada collection olarak tanımlamamız bize bir fayda sağlamayabilir, doğrudan EnumSet gönderilmesini isteyebiliriz

Reply:
Değiştirdim